Index: mergemaster.sh
===================================================================
RCS file: /home/ncvs/src/usr.sbin/mergemaster/mergemaster.sh,v
retrieving revision 1.6.2.17
diff -u -r1.6.2.17 mergemaster.sh
--- mergemaster.sh	10 Mar 2003 06:55:48 -0000	1.6.2.17
+++ mergemaster.sh	20 Apr 2003 23:47:32 -0000
@@ -15,7 +15,7 @@
 display_usage () {
   VERSION_NUMBER=`grep "[$]FreeBSD:" $0 | cut -d ' ' -f 4`
   echo "mergemaster version ${VERSION_NUMBER}"
-  echo 'Usage: mergemaster [-scrvahipC] [-m /path]'
+  echo 'Usage: mergemaster [-scrvahipCP] [-m /path]'
   echo '         [-t /path] [-d] [-u N] [-w N] [-D /path]'
   echo "Options:"
   echo "  -s  Strict comparison (diff every pair of files)"
@@ -27,6 +27,7 @@
   echo '  -i  Automatically install files that do not exist in destination directory'
   echo '  -p  Pre-buildworld mode, only compares crucial files'
   echo '  -C  Compare local rc.conf variables to the defaults'
+  echo '  -P  Preserve files that are overwritten'
   echo "  -m /path/directory  Specify location of source to do the make in"
   echo "  -t /path/directory  Specify temp root directory"
   echo "  -d  Add date and time to directory name (e.g., /var/tmp/temproot.`date +%m%d.%H.%M`)"
@@ -237,7 +238,7 @@
 
 # Check the command line options
 #
-while getopts ":ascrvhipCm:t:du:w:D:" COMMAND_LINE_ARGUMENT ; do
+while getopts ":ascrvhipCPm:t:du:w:D:" COMMAND_LINE_ARGUMENT ; do
   case "${COMMAND_LINE_ARGUMENT}" in
   s)
     STRICT=yes
@@ -269,6 +270,10 @@
   C)
     COMP_CONFS=yes
     ;;
+  P)
+    PRESERVE_FILES=yes
+    PRESERVE_FILES_DIR=/var/tmp/mergemaster-preserved-files-`date +%y%m%d-%H%M%S`
+    ;;
   p)
     PRE_WORLD=yes
     unset COMP_CONFS
@@ -627,6 +632,15 @@
 # Create directories as needed
 #
 do_install_and_rm () {
+  case "${PRESERVE_FILES}" in
+  [Yy][Ee][Ss])
+    if [ -f "${3}/${2##*/}" ]; then
+      mkdir -p ${PRESERVE_FILES_DIR}
+      cp ${3}/${2##*/} ${PRESERVE_FILES_DIR}
+    fi
+    ;;
+  esac
+
   install -m "${1}" "${2}" "${3}" &&
   rm -f "${2}"
 }
