--- pkgdb.rb.ORIG	Sat Nov  2 22:51:20 2002
+++ pkgdb.rb	Sat Nov  2 22:16:16 2002
@@ -309,7 +309,7 @@
 
       PortInfo.match?(pattern, origin) or next
 
-      if pkgnames = deorigin(origin)
+      if origin && (pkgnames = deorigin(origin))
 	ret.concat(pkgnames)
       end
     end
@@ -830,7 +830,7 @@
       if origin = pkg.origin
 	# ..and ports dependencies
 	PortsDB.instance.all_depends_list(origin).each do |o|
-	  if bdeps = deorigin(o)
+	  if o && (bdeps = deorigin(o))
 	    deps.concat(bdeps)
 	  end
 	end
