--- eldoc.el.ORIG	Fri Mar  7 01:55:18 2003
+++ eldoc.el	Fri Mar  7 02:12:05 2003
@@ -275,17 +275,43 @@
        (not (eq (selected-window) (minibuffer-window)))))
 
 
+(defun eldoc-get-doc ()
+  "Return the doc for the current symbol."
+  (let ((current-symbol (eldoc-current-symbol))
+	(current-fnsym  (eldoc-fnsym-in-current-sexp)))
+    (cond ((eq current-symbol current-fnsym)
+	   (or (eldoc-get-fnsym-args-string current-fnsym)
+	       (eldoc-get-var-docstring current-symbol)))
+	  (t
+	   (or (eldoc-get-var-docstring current-symbol)
+	       (eldoc-get-fnsym-args-string current-fnsym))))))
+
 (defun eldoc-print-current-symbol-info ()
   (and (eldoc-display-message-p)
-       (let* ((current-symbol (eldoc-current-symbol))
-              (current-fnsym  (eldoc-fnsym-in-current-sexp))
-              (doc (cond ((eq current-symbol current-fnsym)
-                          (or (eldoc-get-fnsym-args-string current-fnsym)
-                              (eldoc-get-var-docstring current-symbol)))
-                         (t
-                          (or (eldoc-get-var-docstring current-symbol)
-                              (eldoc-get-fnsym-args-string current-fnsym))))))
-         (eldoc-message doc))))
+       (eldoc-message (eldoc-get-doc))))
+
+(defun eldoc-insert-elisp-func-template (doc)
+  "Insert function template extracted from an eldoc help message."
+  (interactive "*")
+  (message "%s" doc)
+  (if (not doc)
+      (error "could not find doc.")
+    (if (string-match "[^(]*(\\(.*\\))[^)]*" doc)
+	(save-excursion
+	  (insert (substring doc (match-beginning 1) (match-end 1)) ")"))
+      (message "Cannot find args, none?"))))
+
+(defun eldoc-doc (&optional insert-template)
+  "Display simple help summary in echo area on demand.
+If INSERT-TEMPLATE is non-nil (interactively with prefix arg) then insert a
+function template at point.
+@todo can we add possibility of specifying what to get help on?"
+  (interactive "P")
+  (let ((doc (eldoc-get-doc)))
+    (if insert-template
+	(eldoc-insert-elisp-func-template doc)
+    (message "%s" (or doc 
+		      (format "No doc for `%s'" (eldoc-current-symbol)))))))
 
 ;; Return a string containing the function parameter list, or 1-line
 ;; docstring if function is a subr and no arglist is obtainable from the
